Source/JavaScriptCore/dfg/DFGClobberize.h

```diff
@@ -1247,6 +1247,20 @@ void clobberize(Graph& graph, Node* node, const ReadFunctor& read, const WriteFu
         def(HeapLocation(CheckTypeInfoFlagsLoc, JSCell_typeInfoFlags, node->child1()), LazyNode(node));
         return;
 
+    case HasStructureWithFlags: {
+        unsigned flags = node->structureFlags();
+        ASSERT(flags);
+
+        if (Structure::bitFieldFlagsCantBeChangedWithoutTransition(flags)) {
+            read(JSCell_structureID);
+            def(HeapLocation(HasStructureWithFlagsLoc, JSCell_structureID, node->child1()), LazyNode(node));
+            return;
+        }
```

Comment:
This is not correct if the Structure is Dictionary.

Reply:
Fixed, thanks!